### PR TITLE
Add Enhancer for using Radium with ES6 classes

### LINKED ISSUE #122: Using Radium with ES6 components
Hi, 

I wrote a class that when inherited allows creating component in an ES6 fashion. This class triggers different parts of Radium.wrap during its lifecycle, so it's quite dirty but here it is :

```javascript
import {Component} from 'react';
import {wrap} from 'radium';

class RadiumComponent extends Component {

    constructor(props, opts) {
        super(props, opts);
        if (this.state) {
            this.state._radiumStyleState = {};
        } else {
            this.state = { _radiumStyleState: {} };
        }
    }

    componentWillUnmount() {
        super.componentWillUnmount();

        if (this._radiumMouseUpListener) {
            this._radiumMouseUpListener.remove();
        }

        if (this._radiumMediaQueryListenersByQuery) {
            Object.keys(this._radiumMediaQueryListenersByQuery).forEach(
                function (query) {
                    this._radiumMediaQueryListenersByQuery[query].remove();
                },
                this
            );
        }
    }

    renderWithStyles() {
        console.error("renderWithStyles should be defined by your component");
        return null;
    }

    render(){
        var renderedElement = this.renderWithStyles();
        return wrap({render: () => renderedElement }).render();
    }
}

export default RadiumComponent ;
```

Your component just needs to implement a renderWithStyles method.
I'm not doing a PR as this is clearly a draft, and I don't know if you're interested in this in the short-term.
Including it in the project would allow for a cleaner implementation, since hacky calls to wrap would not be needed anymore.

Moreover, I don't know if classes and modules works well with JSX-loader.

### PR DESCRIPTION
**Pull request by ianobermiller**
I'm not really sure about the naming conventions, but this is a "higher order component". I've also played around with the idea of calling it a Decorator, since it can be used as-is as an ES7 decorator (https://github.com/wycats/javascript-decorators). We might also consider exposing `wrapUtils` (to satisfy cases like #123), although I'm not sure it is worth expanding the API surface.

Fixes #122.

### CODE REVIEW

**Comment by alexlande** [comment]
Finally played around with this and love it. Feels much nicer than the wrap method. Sorry about the merge conflict I introduced with the `keyframes` thing. :shipit:

**Comment by ianobermiller** [comment]
How about naming? @colinmegill any comments here?

Some options:
- Enhancer
- Decorator
- Wrapper (tad confusing since we have `wrap`, but also maybe consistent?)
- HigherOrderComponent (far too wordy)

**Comment by ianobermiller** [comment]
I'll add more tests for Enhancer once we have babel compilation support; it is hard to do stuff with classes without the class syntax :)

**Comment by ianobermiller** [comment]
I'll merge this now, and we can change the name afterward if we so desire.

**Comment by alexlande** [comment]
Posthumous comment: I like Enhancer. Lines up with a lot of the other higher order component discussions I've seen. `Decorator` implies that it is specifically an ES7 decorator, which would be confusing. `Wrapper` is too similar to `wrap`. `HigherOrderComponent` is indeed too long.
